### PR TITLE
[#2451] set h1-h4 for content-pages

### PR DESCRIPTION
**Pull request by jiromaykin**
Depended on #1196 
Separated this issue off of that preparation.

Taiga: https://taiga.maykinmedia.nl/project/open-inwoner/task/2451

### CODE REVIEW

File: src/open_inwoner/components/templates/components/Form/FileInput.html [hunk 1]
```diff
@@ -17,7 +17,7 @@
     {% if field.errors %}{% errors errors=field.errors %}{% endif %}
 
     <div class="file-list" aria-live="polite">
-        <div class="file-list-selection" hidden><h4 class="file-list-selection__h4">{% if multiple %}{% trans 'Geselecteerde bestanden' %}{% else %}{% trans 'Geselecteerd bestand' %}{% endif %}</h4></div>
+        <div class="file-list-selection" hidden><h4 class="utrecht-heading-3 file-list-selection__heading">{% if multiple %}{% trans 'Geselecteerde bestanden' %}{% else %}{% trans 'Geselecteerd bestand' %}{% endif %}</h4></div>
```
Comment (Bartvaderkin) [inline src/open_inwoner/components/templates/components/Form/FileInput.html:20]: This changed a level 4 to a level 3

File: src/open_inwoner/scss/components/Button/Button.scss [hunk 2]
```diff
@@ -214,7 +214,9 @@
 
 .h2 + .button,
 .h4 + .button,
-.p + .button {
+.p + .button,
+.utrecht-heading-2 + .button,
+.utrecht-heading-4 + .button {
```
Comment (Bartvaderkin) [inline src/open_inwoner/scss/components/Button/Button.scss:219]: Should this set have a level 3?

File: src/open_inwoner/templates/pages/product/form.html [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
 {% block content %}
 
-    <h1 class="h1" id="title">
+    <h1 class="h1 utrecht-heading-1" id="title">
```
Comment (Bartvaderkin) [inline src/open_inwoner/templates/pages/product/form.html:16]: This has both h1 and utrecht class

File: src/open_inwoner/templates/pages/profile/actions/list.html [hunk 4]
```diff
@@ -5,7 +5,7 @@
 {% block content %}
     {% render_grid %}
         {% render_column span=9 %}
-            <h1 class="h1" id="title">
+            <h1 class="h1 utrecht-heading-1" id="title">
```
Comment (Bartvaderkin) [inline src/open_inwoner/templates/pages/profile/actions/list.html:8]: Another double class.

File: src/open_inwoner/templates/pages/profile/contacts/list.html [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 
 {% block content %}
-<h1 class="h1" id="title">
+<h1 class="h1 utrecht-heading-1" id="title">
```
Comment (Bartvaderkin) [inline src/open_inwoner/templates/pages/profile/contacts/list.html:6]: Blip

File: src/open_inwoner/templates/pages/category/detail.html [hunk 6]
```diff
@@ -11,7 +11,7 @@
 
 {% block content %}
     <div class="categories__content">
-        <h1 class="utrecht-heading-1">
+        <h1 class="h1 utrecht-heading-1">
```
Comment (Bartvaderkin) [inline src/open_inwoner/templates/pages/category/detail.html:14]: This has both old and new class.

File: src/open_inwoner/templates/pages/plans/list.html [hunk 7]
```diff
@@ -6,7 +6,7 @@
 {% endblock header_image %}
 
 {% block content %}
-    <h1 class="h1">
+    <h1 class="h1 utrecht-heading-1">
```
Comment (Bartvaderkin) [inline src/open_inwoner/templates/pages/plans/list.html:9]: Double class

**Review by Bartvaderkin — CHANGES_REQUESTED** [review]
Did you touch every template? :wink:

File: src/open_inwoner/components/templates/components/Form/FileInput.html (hunk 1 above)
Comment (Bartvaderkin) [reply src/open_inwoner/components/templates/components/Form/FileInput.html:20]: Oh I noticed this matches the SCCS change later on.

Comment (jiromaykin) [reply src/open_inwoner/components/templates/components/Form/FileInput.html:20]: Yeah :-) this is a situation where the level of the tag stays the same, but it's just the styling that changes. So here the H4 needs to look like an H3. So I'm keeping this like this.

File: src/open_inwoner/scss/components/Button/Button.scss (hunk 2 above)
Comment (jiromaykin) [reply src/open_inwoner/scss/components/Button/Button.scss:219]: No, these are slightly annoying 'context' styles anyway, so the less, the better. 
Styles for margins should actually be set by the parent, not by the component itself (but... for content on pages it does feel a bit odd to call the page-content 'the parent'). 
So I've put these in here, as a copy of the 'legacy' styling.

File: src/open_inwoner/templates/pages/product/form.html (hunk 3 above)
Comment (jiromaykin) [reply src/open_inwoner/templates/pages/product/form.html:16]: I'm not sure if this template is actually used in practice anywhere but I'll change it anyway 😬

**Comment by jiromaykin** [comment]
> Did you touch every template? 😉

Brace yourself for more NLDS to come...

**Review by pi-sigma — CHANGES_REQUESTED** [review]
The templates:

- `registration/password_reset_confirm.html`
- `accounts/email_verification.html`

still have `<h1 class="h1">`. Are they are outside the scope of this PR?

**Comment by jiromaykin** [comment]
> * `registration/password_reset_confirm.html`
> * `accounts/email_verification.html`
> still have `<h1 class="h1">`. Are they are outside the scope of this PR?

Ah, I thought those were just email templates, but I see they are indeed front-end so I will change those too.

**Review by pi-sigma — APPROVED** [review]

**Review by Bartvaderkin — APPROVED** [review]

**Pull request merged**